src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -0,0 +1,806 @@
+        auto q_len = query.size(2);
+        auto S = query.size(3);
+        auto kv_len = present_key.size(2);
+        auto h_group_num = present_key.size(1);
+        size_t h_each_group_len = 0;
+        if (h_group_num != H) {
+            h_each_group_len = H / h_group_num;
+        }
+
+        if (d_scale == 0.0f)
+            d_scale = 1.0f / sqrt(S);
+
+        // use per-token kernel, for each k,v token
+        //  attn mask is a matrix of q_len(kv_len)
+        m_attn_w.resize({B, H, q_len, kv_len});
+
+        if (h_each_group_len) {
+            parallel_for3d(B, H / h_each_group_len, kv_len, [&](size_t b, size_t h_group, size_t pk) {
+                // which batch item should be used at postion pk?
+                auto b_kv = beams ? beams.at({b, pk}) : b;
+                for (size_t pq = 0; pq < q_len; pq++) {
+                    for (size_t h = h_group * h_each_group_len; h < (h_group + 1) * h_each_group_len; h++) {
+                        auto sum = attn_dot_product(&query.at({b, h, pq, 0}),
+                                                          &present_key.at({b_kv, h_group, pk, 0}),
+                                                        S,
+                                            precision_of<RT>::value);
+                        m_attn_w.at({b, h, pq, pk}) = sum;
+                    }
+                }
+            });
+        } else {
+            parallel_for3d(B, H, kv_len, [&](size_t b, size_t h, size_t pk) {
+                // which batch item should be used at postion pk?
+                auto b_kv = beams ? beams.at({b, pk}) : b;
+                for (size_t pq = 0; pq < q_len; pq++) {
+                    auto sum = attn_dot_product(&query.at({b, h, pq, 0}),
+                                                      &present_key.at({b_kv, h, pk, 0}),
+                                                      S,
+                                                      precision_of<RT>::value);
+                    m_attn_w.at({b, h, pq, pk}) = sum;
+                }
+            });
+        }
+
+        parallel_for3d(B, H, q_len, [&](size_t b, size_t h, size_t pq) {
+            // apply attention mask & sofmax
+            auto ncausal = auto_causal ? (kv_len - q_len + pq + 1) : kv_len;
+            float* alibi_ptr = alibi_mask ? &alibi_mask.at({b, h, pq, 0}, true) : nullptr;
+            float* attn_mask_ptr = attention_mask ? &attention_mask.at({b, h, pq, 0}, true) : nullptr;
+            uint8_t* cmask_ptr = causal_mask ? &causal_mask.at({b, h, pq, 0}, true) : nullptr;
+            attn_softmax(&m_attn_w.at({b, h, pq, 0}),
+                         &m_attn_w.at({b, h, pq, 0}),
+                         d_scale,
+                         alibi_ptr,
+                         attn_mask_ptr,
+                         cmask_ptr,
+                         select_nfltmax_at_0,
+                         ncausal,
+                         kv_len,
+                         Precision::FP32);
```

Comment:
`cross_compiled_file` for this function is OK because it's not inside a loop, wrapper overhead is not  a big deal. but I'm not sure whether it's the case or not for `attn_dot_product` and `attn_acc_value`

Reply:
Move the loop into `attn_dot_product` and `attn_acc_value`.